### PR TITLE
[KOGITO-844] Replacing VFSService with ResourceContentService

### PR DESCRIPTION
**Pull request by gitgabrio**
See https://issues.redhat.com/browse/KOGITO-844
This PR is needed to avoid propagation of backend modules to kogito editors.

### CODE REVIEW

**Comment by gitgabrio** [comment]
jenkins execute full downstream build

**Comment by manstis** [comment]
@gitgabrio In addition to my comments in the review it might be worth changing the constructor for `KogitoResourceContentService` to the following (so you can return `promises.resolve()` instead of `null` and `SyncPromises` in the test).
```
@Inject
public KogitoResourceContentService(final PlaceManager placeManager,
                                    final ResourceContentService resourceContentService,
                                    final Promises promises) {
    this.placeManager = placeManager;
    this.resourceContentService = resourceContentService;
    this.promises = promises;
}
```

**Comment by tiagobento** [comment]
@manstis Regarding Promises, think of it as a `try/catch/finally` block. You don't (usually) use the `catch` block as an `else` statement, instead, you return an object and act upon its value. Same thing applies to Promises.

About the reject callback VS the catch block, they're the same thing. I think this API might exist on some specification and elemental2 generated it. I usually prefer using the catch block for handling errors/exceptions.

**Comment by gitgabrio** [comment]
@manstis @jomarko 
Just fixed as per @manstis suggestion.

**Comment by gitgabrio** [comment]
jenkins please retest this

**Comment by gitgabrio** [comment]
> This PR does not work well in conjunction with #3103
> 
> ```
> [INFO] --- gwt-maven-plugin:2.8.2:compile (gwt-compile) @ kie-wb-common-dmn-webapp-kogito-runtime ---
> [INFO] Compiling module org.kie.workbench.common.dmn.showcase.DMNKogitoRuntimeWebapp
> [INFO]    Tracing compile failure path for type 'org.kie.workbench.common.kogito.webapp.base.client.workarounds.KogitoResourceContentService'
> [INFO]       [ERROR] Errors in 'jar:file:/home/manstis/.m2/repository/org/kie/workbench/kie-wb-common-kogito-webapp-base/7.32.0-SNAPSHOT/kie-wb-common-kogito-webapp-base-7.32.0-SNAPSHOT-sources.jar!/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java'
> [INFO]          [ERROR] Line 42: No source code is available for type org.appformer.kogito.bridge.client.resource.ResourceContentService; did you forget to inherit a required module?
> [INFO]    [ERROR] Aborting compile due to errors in some input files
> ```
> 
> May I suggest this PR be re-factored and re-reviewed when the former has been merged?

@manstis This one is mandatory (blocking) to proceed with scesim kogito (https://issues.jboss.org/browse/DROOLS-3879), while #3103 is not necessary (I started it later) for DMN to work inside Kogito, so I would prefer to have this one merged before, if possible.

File: kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                                        final Promises promises) {
+        this.placeManager = placeManager;
+        this.resourceContentService = resourceContentService;
+        this.promises = promises;
+    }
+
+    /**
+     * Open a file at given <code>Path</code> inside an <b>editor</b>
+     * @param fileUri the <b>uri</code> to the file
+     * @param editorId The <b>id</b> of the editor to open by the <code>PlaceRequest</code>
```
Comment (jomarko) [inline kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java]: So is the first parameter `uri` or `path`? I am confused after reading javadoc.

File: kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            placeManager.goTo(placeRequest);
+            return promises.resolve();
+        });
+    }
+
+    /**
+     * Load the file at given <code>Path</code> and returns content inside a callback
+     * @param fileUri
+     * @param callback
+     */
```
Comment (jomarko) [inline kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java]: similar question uri vs path

File: kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                         final RemoteCallback<String> callback,
+                         final ErrorCallback<Object> errorCallback) {
+        resourceContentService.get(fileUri).then((IThenable.ThenOnFulfilledCallbackFn<String, Void>) fileContent -> {
+                                                     callback.callback(fileContent);
+                                                     return promises.resolve();
+                                                 })
+                .catch_(error -> {
+                    errorCallback.error("Error " + error, new Throwable("Failed to load file " + fileUri));
+                    return null;
+                })
```
Comment (jomarko) [inline kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java]: Don't we need something similar for opening file?

File: kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        ;
+    }
+
+    /**
+     * Get the <code>List&lt;Path&gt;</code> from the project/workspace where the editor is running
```
Comment (jomarko) [inline kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java]: Can we clarify the javadoc and state that obtained list is returned via callback. As the method returns `void`?

File: kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+     *
+     * @param callback
+     * @param errorCallback
+     *
+     * @see ResourceContentService#list(String)
+     */
+    public void getAllItems(final RemoteCallback<List<String>> callback,
+                               final ErrorCallback<Object> errorCallback) {
```
Comment (jomarko) [inline kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java]: formatting

File: kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    /**
+     * Get <b>filtered</b> <code>List&lt;Path&gt;</code> from the project/workspace where the editor is running
+     * @param pattern A GLOB pattern to filter files. To list all files use "*"
+     * @param callback
+     * @param errorCallback
+     *
+     * @see ResourceContentService#list(String)
+     */
```
Comment (jomarko) [inline kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java]: similar question as above

File: kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    public void getFilteredItems(final String pattern,
+                               final RemoteCallback<List<String>> callback,
+                               final ErrorCallback<Object> errorCallback) {
```
Comment (jomarko) [inline kie-wb-common-kogito/kie-wb-common-kogito-webapp-base/src/main/java/org/kie/workbench/common/kogito/webapp/base/client/workarounds/KogitoResourceContentService.java]: formating

**Comment by gitgabrio** [comment]
@yesamer @jomarko 
Fixed as per your reviews. Could you please check again?

**Pull request merged**